app/src/main/java/org/thoughtcrime/securesms/contactshare/ContactFieldAdapter.java

```diff
@@ -0,0 +1,236 @@
+
+    private final TextView  value;
+    private final TextView  label;
+    private final ImageView icon;
+    private final ImageView avatar;
+    private final CheckBox  checkBox;
+
+    ContactFieldViewHolder(View itemView) {
+      super(itemView);
+
+      value    = itemView.findViewById(R.id.contact_field_value);
+      label    = itemView.findViewById(R.id.contact_field_label);
+      icon     = itemView.findViewById(R.id.contact_field_icon);
+      avatar   = itemView.findViewById(R.id.contact_field_avatar);
+      checkBox = itemView.findViewById(R.id.contact_field_checkbox);
+    }
+
+    void bind(@NonNull Field field, @NonNull GlideRequests glideRequests, boolean selectable) {
+      value.setMaxLines(field.maxLines);
+      value.setText(field.value);
+      label.setText(field.label);
+      icon.setImageResource(field.iconResId);
+
+      if (field.iconUri != null) {
+        avatar.setVisibility(View.VISIBLE);
+        glideRequests.load(field.iconUri)
+                     .diskCacheStrategy(DiskCacheStrategy.NONE)
+                     .skipMemoryCache(true)
+                     .circleCrop()
+                     .into(avatar);
+      } else {
+        avatar.setVisibility(View.GONE);
+      }
+
+      if (selectable) {
+        checkBox.setVisibility(View.VISIBLE);
+        checkBox.setOnCheckedChangeListener(null);
+        checkBox.setChecked(field.isSelected());
+        checkBox.setOnCheckedChangeListener((buttonView, isChecked) -> field.setSelected(isChecked));
+        super.itemView.setOnClickListener(v -> checkBox.toggle());
+      } else {
+        checkBox.setVisibility(View.GONE);
+        checkBox.setOnCheckedChangeListener(null);
+      }
+    }
+
+    void recycle() {
+      checkBox.setOnCheckedChangeListener(null);
+    }
+  }
+
+  static class Field {
+
+    final String     value;
+    final String     label;
+    final int        iconResId;
+    final int        maxLines;
+    final Selectable selectable;
+
+    @Nullable
+    final Uri        iconUri;
+
+    Field(@NonNull Context context, @NonNull Phone phoneNumber, @NonNull Locale locale) {
+      this.value      = ContactUtil.getPrettyPhoneNumber(phoneNumber, locale);
+      this.iconResId  = R.drawable.ic_phone_right_unlock_solid_24;
+      this.iconUri    = null;
+      this.maxLines   = 1;
+      this.selectable = phoneNumber;
+
+      switch (phoneNumber.getType()) {
+        case HOME:
+          label = context.getString(R.string.ContactShareEditActivity_type_home);
+          break;
+        case MOBILE:
+          label = context.getString(R.string.ContactShareEditActivity_type_mobile);
+          break;
+        case WORK:
+          label = context.getString(R.string.ContactShareEditActivity_type_work);
+          break;
+        case CUSTOM:
+          label = phoneNumber.getLabel() != null ? phoneNumber.getLabel() : "";
+          break;
+        default:
+          label = "";
+      }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 154:169, 179:194.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.